### PR TITLE
SMTP Server Availability Check

### PR DESCRIPTION
**Pull request by mjdobson**
It's a simple script that checks the availability of an SMTP Server by connecting and evaluating the response code.  I improved on some code Bernie Clairmont provided per case S-CS-0164088 such that the script will fail if the response code is not as expected.

### CODE REVIEW

**Comment by mjdobson** [comment]
Could you review @hashlock-te and @benton-te ?

File: examples/smtpServerAvailability.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* 
+    A script that checks availability of an SMTP Server by connecting to the service and evaluating the response code.
+*/
+
+import { By, Key, until } from 'selenium-webdriver';
```
Comment (benton-te) [inline examples/smtpServerAvailability.js]:
```suggestion
```

File: examples/smtpServerAvailability.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { net, driver, markers, credentials, downloads, transaction, authentication } from 'thousandeyes';
```
Comment (benton-te) [inline examples/smtpServerAvailability.js]:
```suggestion
import { net } from 'thousandeyes';
```

File: examples/smtpServerAvailability.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import assert from 'assert';
+runScript();
+
+async function runScript() {
+    // Set host and port
+    let host = 'smtp.gmail.com';
+    let port = 587;
+    let resp = await callServer(host, port);
```
Comment (benton-te) [inline examples/smtpServerAvailability.js]:
```suggestion
    await callServer(host, port);
```
This function does not return anything, and the variable resp is not used anyway. No need to assign to a variable

File: examples/smtpServerAvailability.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    let response = await sock.read();
+    await sock.writeAll('HELO ' + host + '\r\n');
+    response = await sock.read();
+    // Check entire response with option below when using IDE
+    //console.log (response);
+    await checkCode(response,250);
+    await sock.writeAll('QUIT\r\n');
+    response = await sock.read();
+}
+
```
Comment (benton-te) [inline examples/smtpServerAvailability.js:35]: You might consider adding comments for each of the events to clarify what's going on. And I would recommend a blank line before each comment to improve readability.

File: examples/smtpServerAvailability.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    const sock = await net.connect(port, host);
+    sock.setEncoding('utf8');
+    let response = await sock.read();
+    await sock.writeAll('HELO ' + host + '\r\n');
+    response = await sock.read();
+    // Check entire response with option below when using IDE
+    //console.log (response);
+    await checkCode(response,250);
+    await sock.writeAll('QUIT\r\n');
+    response = await sock.read();
```
Comment (benton-te) [inline examples/smtpServerAvailability.js]:
```suggestion
    response = await sock.read();
    await sock.end()
```
Just to be nice to the server

Comment (mjdobson) [reply examples/smtpServerAvailability.js]: haha, oops!

**Review by benton-te — APPROVED** [review]
LGTM

**Pull request merged**